### PR TITLE
Fix fail unit tests old node versions

### PR DESCRIPTION
**Pull request by AllanFly120**
fix 2 issues that fail the TravisCI testing:
1. An issue in unit test brought in #1880. In current version of node, `Hmac.digest` will return a buffer. But in node <= 0.12, `Hmac.digest` can return `Dataview` and it's not printable. So the test failed by asserting '[object Uint8Array]' to equal a binary string. Fix it by manually construct buffer for this kind of output.
2. A compatible issue brought in #1905. This leads to `sharedInitFileCredentials` cannot load chaining role ARN in Node before 4.0. The `Object.assign` is not available in old JS we support. Fix it by using `AWS.util.merge`

### CODE REVIEW

File: test/util.spec.js [hunk 1]
```diff
@@ -688,16 +688,28 @@
...
       obj = {
         a: 1,
         b: 2
       };
       return expect(AWS.util.update(obj, {
         c: 3
-      })).to.equal(obj);
+      })).to.eql({a: 1, b: 2, c: 3});
     });
+    it('should merge an object into undefined object', function() {
```
Comment (chrisradek) [inline test/util.spec.js]: Why should this work? I don't think updating an object that is `undefined` or `null` is expected, since this method is currently meant to mutate an existing object. Was this change needed to fix something?

File: test/browserHashes.spec.js [hunk 2]
```diff
@@ -43,7 +43,11 @@ describe('Browser hash implementations', function() {
                         if (truncate) {
                             digest = digest.slice(0, truncate);
                         }
-                        expect(digest.toString('hex')).to.equal(expected);
+                        //in node <= 0.10 digest sometimes returns a Dataview, should be buffer.
```
Comment (chrisradek) [inline test/browserHashes.spec.js]: Are these tests running in node.js? This should only be running in browser environments, and is using the 3rd party `Buffer` package instead of node.js' `Buffer` package. We can place browser-specific tests in a separate folder and exclude them from being run by mocha in the npm unit script.

Comment (AllanFly120) [reply test/browserHashes.spec.js]: Ah right! The change mean to fix the test. But actually we can remove it out of node test. I didn't realize this is browser-only although the name already told me

File: test/util.spec.js (hunk 1 above)
Comment (AllanFly120) [reply test/util.spec.js]: [here](https://github.com/aws/aws-sdk-js/blob/214b42460882384ab1ac5edd6ec48a6076f31f8f/lib/credentials/shared_ini_file_credentials.js#L181) is where is issue is and I changed to [`AWS.util.merge`](https://github.com/aws/aws-sdk-js/blob/214b42460882384ab1ac5edd6ec48a6076f31f8f/lib/util.js#L508). Actually this fix doesn't make sense, instead we should allow merge function to merge undefined or null
